lib/engine/game/g_18_uruguay/game.rb

```diff
@@ -403,6 +416,59 @@ def or_round_finished
         def final_operating_round?
           @final_turn == @turn
         end
+
+        def place_home_token(corporation)
+          return if corporation == @fce
+
+          super
+        end
+
+        def can_hold_above_corp_limit?(_entity)
+          true
+        end
+
+        def next_round!
+          @round =
+            case @round
+            when G18Uruguay::Round::Nationalization
+              case @round.round_num
+              when 1
+                start_merge(current_entity.owner)
+              when 2
+                retreive_home_tokens
+                close_companies
+                @crowded_corps = nil
+                @cert_limit = CERT_LIMIT_NATIONALIZATION[@players.size][@corporations.size]
+                @log << "New certification limit is #{@cert_limit}"
+              end
+
+              if @round.round_num < 3
+                new_nationalization_round(@round.round_num + 1)
+              elsif @saved_or_round
+                # reorder_players
+                @log << '--Return to Operating Round--'
+                @saved_or_round
+              else
+                new_operating_round
+              end
+            when Engine::Round::Operating
+              if @nationalization_triggered
+                @nationalization_triggered = false
+                @saved_or_round = @round
+                new_nationalization_round(1)
+              else
+                super
+              end
+            else
+              super
```

Comment:
Unless you strongly think I should duplicate the code I think I will skip. It might be slightly simplier to follow but as you stated it will be duplications and if there are an update in the base it will not be getting to this part